homekit/controller/tools.py

```diff
@@ -166,3 +173,138 @@ def check_convert_value(val, target_type):
         except (binascii.Error, TlvParseException):
             raise FormatError('"{v}" is no valid "{t}"!'.format(v=val, t=target_type))
     return val
+
+
+class DelayedExecution(threading.Thread):
```

Comment:
My branch contains a backport of a trivial upstream GATT PR that adds a GObject idle timeout. This is only a line of python and avoids threads.